lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -299,6 +299,22 @@ def test_with_climatology(self):
                 saver.write(cube)
             self.assertCDL(nc_path)
 
+    def test_dimensional_to_scalar(self):
```

Comment:
I'm finding that this test did not **_fail_** for me with iris 3.1.0, which it really ought to.
I may have got something wrong, but assuming not : can we fix that ??

Reply:
I introduced the bug in Iris 3.2. Before that we were not lazily streaming bounds data, so never encountered the problem.

Reply:
... and I believe it would be impossible to replicate for `Cube.data` itself, due to the specific ways we handle that, so there's no test that can be written to fail for versions <`v3.2`.

Reply:
OK sorry, I will check this again with latest-master ...

Reply:
Also: I expect this to always pass when run within `test_Saver.Test_write`, but without the fix it should fail as part of `test_Saver__lazy.Test_write`. Does that make sense?

Reply:
Apologies, I see now that just running the new version of `test_Saver.py` does not exercise the correct test.
It is essential that I _replace_ `test_Saver.py` with the new one, and then run `test_Saver__lazy.py`.
Which is a bit subtle + I totally missed it !

Reply:
Ah ok!
I think I hadn't seen that there were 2 instances of this, and was running the wrong one ...